sky/provision/aws/config.py

```diff
@@ -195,7 +195,25 @@ def _get_role(role_name: str):
     return {'Arn': profile.arn}
 
 
+def _is_subnet_public(ec2, subnet_id) -> bool:
```

Comment:
Could you clarify? It seems that public subnets conventionally connect to a IGW, and private subnets to a NAT: https://docs.aws.amazon.com/vpc/latest/userguide/vpc-example-private-subnets-nat.html

Re naming: I feel the current name places less burden on the callers who do not need to understand how the func determines if a subnet is considered public. Wdyt?

Reply:
Ahh, I was confused between the IGW and NAT. Can we have a reference link in the comment to show that IGW is used by subnet with public IPs, such as https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Internet_Gateway.html
> An internet gateway enables resources in your public subnets (such as EC2 instances) to connect to the internet if the resource has a public IPv4 address or an IPv6 address. Similarly, resources on the internet can initiate a connection to resources in your subnet using the public IPv4 address or IPv6 address. For example, an internet gateway enables you to connect to an EC2 instance in AWS using your local computer.

Renaming: I think the current name works well. I was confused because I thought the IGW can be used by private subnet but it seems not.

Reply:
Added.